### PR TITLE
add dependendies to run on Debian Bullseye

### PR DESCRIPTION
**Pull request by hansu**
The python package `pyopengl` is also required. But I don't know how to add this.

### CODE REVIEW

**Comment by petterreinholdtsen** [comment]
[Hans Unzner]
> The python package `pyopengl` is also required. But I don&#39;t know
> how to add this.

Which error is this patch solving?

I suspect you can add extra binary dependencies in either
debian/control.main-pkg.in or debian/configure.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by hansu** [comment]
>Which error is this patch solving?

Building and running LinuxCNC 2.9 on Bullseye

File: debian/control.top.in [hunk 1]
```diff
@@ -23,10 +23,12 @@ Build-Depends:
     intltool,
     libboost-python-dev,
     libepoxy-dev,
+    libghc-glut-dev,
```
Comment (smoe) [inline debian/control.top.in:26]: That is for the haskell environment. I'd opt for libglut-dev, like
```
libglut-dev|freeglut3-dev
```
which should then work for both bullseye and bookworm (the latter still missing that symbolic link, but we'll eventually get there).

Comment (hansu) [reply debian/control.top.in:26]: `libglut-dev` is not available in the Bullseye sources, but  `freeglut3` or `freeglut3-dev` do work as well.
So 
```diff
- libghc-glut-dev,
+ freeglut3|freeglut3-dev
```
 ?

**Comment by SebKuzminsky** [comment]
2.9 builds fine for me on Bullseye.  I don't see why this patch is needed.  What am I missing?

**Comment by hansu** [comment]
Did you also use the netinstall ISO?

File: debian/control.top.in (hunk 1 above)
Comment (smoe) [reply debian/control.top.in:26]: freeglut3-dev is fine for bullseye but for bookworm this is only transitional, libglut-dev is what we should use for the latter, so `libglut-dev|freeglut3-dev` is fine.

**Comment by smoe** [comment]
I am still uncertain if we should specify that dependency for linuxcnc-uspace directly. The python3-opengl package should provide all that is needed at runtime.

**Comment by SebKuzminsky** [comment]
@hansu is this PR still relevant, or can we close it?  As far as I can tell, and according to both Github Actions and the buildbot, linuxcnc 2.9 builds fine on Bullseye.

**Comment by hansu** [comment]
This is not (only) about building, more than about running the build, see https://github.com/LinuxCNC/linuxcnc/pull/2275#issuecomment-1384917439.
Maybe the title was a bit imprecise, I changed it.

**Comment by smoe** [comment]
Anything that is needed for the execution is a runtime dependency, not a build-dependency, so the patch needs to be adjusted for that. And "ghc" is the haskell compiler (https://www.haskell.org/ghc/) - would like to avoid depending on that since we do not have any haskell bindings.

**Comment by smoe** [comment]
@hansu, maybe you rebase and push again? Something seems not to have arrived since the patch is still against the build-deps.

**Comment by hansu** [comment]
> @hansu, maybe you rebase and push again? Something seems not to have arrived since the patch is still against the build-deps.

Yeah that's because I haven't changed anything here yet :innocent: 

So do you mean I should add runtime dependencies under `Depends` - is that the right place?

**Comment by SebKuzminsky** [comment]
Please open a new PR when the issues with this PR have been addressed.

**Comment by hansu** [comment]
@SebKuzminsky @smoe 
The thing is that `python3-opengl` is needed to run on Bullseye. That is the only thing that needs to be installed manually when trying to run a rip.

And it is not being installed when linuxcnc is installed from a deb package:
```
cnc@bullseye:~/Downloads$ sudo dpkg -i linuxcnc-uspace_2.9.0~pre1.714.g1fd27d043_amd64.deb 
(Reading database ... 292663 files and directories currently installed.)
Preparing to unpack linuxcnc-uspace_2.9.0~pre1.714.g1fd27d043_amd64.deb ...
Unpacking linuxcnc-uspace (1:2.9.0~pre1.714.g1fd27d043) over (1:2.9.0~pre1.714.g1fd27d043) ...
dpkg: dependency problems prevent configuration of linuxcnc-uspace:
 linuxcnc-uspace depends on python3-opengl; however:
  Package python3-opengl is not installed.
 linuxcnc-uspace depends on libgtksourceview-3.0-dev; however:
  Package libgtksourceview-3.0-dev is not installed.
 linuxcnc-uspace depends on tclreadline; however:
  Package tclreadline is not installed.

dpkg: error processing package linuxcnc-uspace (--install):
 dependency problems - leaving unconfigured
Processing triggers for libc-bin (2.31-13+deb11u5) ...
Processing triggers for gnome-menus (3.36.0-1) ...
Processing triggers for desktop-file-utils (0.26-1) ...
Processing triggers for mailcap (3.69) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for man-db (2.9.4-2) ...
Processing triggers for shared-mime-info (2.0-1) ...
Errors were encountered while processing:
 linuxcnc-uspace
```


How can this be fixed?

**Comment by SebKuzminsky** [comment]
> ```
>  linuxcnc-uspace depends on python3-opengl; however:
>   Package python3-opengl is not installed.
> ```

Looks like linuxcnc-uspace knows that it needs python3-opengl already.  The dpkg tool does not try to install missing dependencies.  Try running `apt install ./linuxcnc-uspace*.deb` instead, apt knows how to install missing dependencies.

**Comment by hansu** [comment]
>Try running apt install ./linuxcnc-uspace*.deb instead, apt knows how to install missing dependencies.

Okay that did work, thanks for the tip.

But for the rip installation it would be nice if `dpkg-checkdeps.rb` would work like `dpkg-checkbuilddeps`.

**Pull request closed without merging**